smarts/core/agent.py

```diff
@@ -92,10 +102,9 @@ class AgentSpec:
     """A callable to build an `smarts.core.agent.Agent` given `AgentSpec.agent_params` (default None)"""
     agent_params: Optional[Any] = None
     """Parameters to be given to `AgentSpec.agent_builder` (default None)"""
-
-    observation_adapter: Callable = lambda obs: obs
+    observation_adapter: Optional[Adapter] = lambda obs: obs
```

Comment:
I am wary about keeping the `lambda` here since it does not follow the typing now specified. I think we should explicitly use `Adapter` or `None`.